### PR TITLE
Playground headless

### PR DESCRIPTION
**Pull request by mthl**
This pull-request allows running a simulation without opening a Qt/OpenGL window.  This is great because now it is possible to run a simulation on a machine without a graphical interface.

The 2 commits represents two iterative steps.  The first version still use a Qt timer but without requiring a GUI.  The second removes the dependency to Qt by using a endless loop with portable C++11 `thread::sleep_for`. I think it is important to keep the two versions in the history to be able to easily roll-back to the Qt version if a regression is observed in the future.

### CODE REVIEW

**Comment by mthl** [comment]
to quickly test if it's works for you, from you build directory you can:
```sh
./viewer/enkiplaygroundHeadless &
./viewer/enkiplaygroundNetwork 127.0.0.1
```

File: viewer/PlaygroundHeadless.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	Thymio2 *thymio = new Thymio2;
+	thymio->leftSpeed = 3;
+	thymio->rightSpeed = 4;
+	thymio->pos = Point(0, 0);
+	world->addObject(thymio);
+
+	Server* server = new Server(world);
+
+	std::thread network([=]() { server->run(); });
```
Comment (nicpalard) [inline viewer/PlaygroundHeadless.cpp]: Can you explain me what this means ?

Comment (mthl) [inline viewer/PlaygroundHeadless.cpp]: this statement creates a thread object called `network` on the stack (meaning without using `new`), and the constructor takes a lambda expression as parameter (think anonymous function).  This function is then called inside the newly created thread.

To describe the semantic of this particular lambda.  It is a function that take no argument and return nothing.  It captures the `server` variable by value (this is the meaning of the `=` inside the `[]`) and call its `run` method.

Does it help?

**Comment by mthl** [comment]
I have updated last commit to benefit from the std::chrono features

**Comment by mthl** [comment]
in this particular case there is not much benefit, but in general allocating objects on the stack is a good practice in C++ since it allows [RAII](https://fr.wikipedia.org/wiki/Resource_acquisition_is_initialization) which simplifies the memory management (compared to using `new`).

**Pull request closed without merging**